### ISSUE TITLE
Unable to get property 'isProxied' of undefined or null reference (like #1661)

### ISSUE THREAD

**Issue by DaanVanVugt** [ionic-team/stencil#2774]
Stencil version:

 @stencil/core@1.0.6
@Stencil/router 1.0.1

I'm submitting a:

[ x ] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://stencil-worldwide.herokuapp.com/ or https://forum.ionicframework.com/

Current behavior:

Probably related to #1516. stencil build --es5 when using Ionic components builds, but then when served produces the error in the console: "Unhandled promise rejection TypeError: Unable to get property 'isProxied' of undefined or null reference"

Expected behavior:

For the PWA toolkit to load in both dev and prod builds.

Steps to reproduce:

The following code is causing the issue, copypasted from the pwa toolkit. Without it, the page renders fine. Please note I am using lite-server, not the Stencil dev server, because I cannot get that to work right now (reported under separate issue).

Related code:

```tsx
    return [
      <ion-header>
        <ion-toolbar color="primary">
          <ion-title>Home</ion-title>
        </ion-toolbar>
      </ion-header>,

      <ion-content class="ion-padding">
        <p>
          Welcome to the PWA Toolkit. You can use this starter to build entire
          apps with web components using Stencil and ionic/core! Check out the
          README for everything that comes in this starter out of the box and
          check out our docs on <a href="https://stenciljs.com">stenciljs.com</a> to get started.
        </p>

        <ion-button href="/profile/ionic" expand="block">Profile page</ion-button>
      </ion-content>
    ];
  }
```

Other information:

Original issue watchers:
@Aaron-Sterling @d0whc3r @meriturva @rieshy @warapitiya @JimGaleForce @havard024 @ali-master @andersonfvrj @pwespi @dutscher

**Comment by coconnorco** [comment]
We've been seeing this error too but on @stencil/core: 2.3.0 (we're not using the pwa toolkit). A vast majority of these errors seem to be coming from safari users but intermittently.

**Comment by DaSchTour** [comment]
I'm getting this error with my published components when trying to load from unpkg. It's very really annoying to have a broken library published and clients waiting for a new version while I have no idea how to fix it. BTW I also have this error in Chrome.

**Comment by DaSchTour** [comment]
I also discovered, that the generated build and documentation is broken, not matching or missleading. In this documentation https://stenciljs.com/docs/distribution the script tag points to `https://cdn.jsdelivr.net/npm/my-name@0.0.1/dist/myname.js`

There are two mistakes in this.
1. There is no file with the library name generated in dist but only an index.js
2. The index.js then points to an empty file

Here you can check my broken published bundle https://www.jsdelivr.com/package/npm/@footage.one/search-components?path=dist

**Comment by ajgagnon** [comment]
I've narrowed down this issue to Safari iOS version 12. The only way to fix this is to run a babel rollup plugin (which breaks tests)

**Comment by splitinfinities** [comment]
Hey folks! Could you please update to the latest version and help me ID if this is intermittent or not? With sourcemaps now shipped in v2.9.0, I hope that folks hitting this can use them to determine where precisely this occurs and why.

**Comment by gcaven-nbx** [comment]
I'm still seeing this issue on v2.12.0, intermittently and mostly (though not exclusively) with mobile Safari users. Up to and including Safari 15.

**Comment by andrewlw89** [comment]
> Getting this issue in Chrome while using a Cypress-Test 🤔

Same issue but with Chrome and Testcafe

**Comment by Clovel** [comment]
I've got this issue on iOS 15 as well. Has anybody found a workaround or a fix ?

**Comment by FabianEbert** [comment]
I am also interested in a fix/workaround. Using our application in Safari is blocked by this issue. Interestingly the application shows up after reloading the page. So I am wondering about an ESM loading issue in WebKit (see below).

After enabling source mapping, I can see that the affected code is [this one](https://github.com/ionic-team/stencil/blob/63dbb47a14cc840c8d37f1bf7ce315d306194788/src/runtime/initialize-component.ts#L44). The implementation dynamic module loading in Safari might have an issue, [this line](https://github.com/ionic-team/stencil/blob/63dbb47a14cc840c8d37f1bf7ce315d306194788/src/runtime/initialize-component.ts#L32) does not seem to work in Safari.

Wondering if is related to this [WebKit issue](https://bugs.webkit.org/show_bug.cgi?id=198665) (Race condition makes ESM modules to evaluate twice)?

Does anyone have more information? Any suggestions are appreciated, thank you.

**Comment by rahulgupta-dev** [comment]
I am also getting error when ever there is new build, can we get some solution or workaround.

p-e82b5571.system.js:1  Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'isProxied')
    at Object.<anonymous> (p-e82b5571.system.js:1:15976)
    at o (p-e82b5571.system.js:1:1866)
    at Object.next (p-e82b5571.system.js:1:1157)
    at l (p-e82b5571.system.js:1:693)

**Comment by rwaskiewicz** [comment]
Hey there 👋 

I’m having trouble reproducing this on my end with the latest version of Stencil (v4). If anyone is willing/able, would you be able to create a minimal reproduction case for the team? If not, I completely understand. Again, I apologize for the lack of response here
